core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.ts

```diff
@@ -0,0 +1,262 @@
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      new_member_role: newMemberRole,
+      new_member_username: newMemberUsername
+    }).toPromise().then((response: ExplorationRightsBackendData) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      this.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+
+  setViewability(
+      viewableIfPrivate: boolean): Promise<void> {
+    let requestUrl = (
+      '/createhandler/rights/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      version: this.explorationDataService.data.version,
+      viewable_if_private: viewableIfPrivate
+    }).toPromise().then((response: ExplorationRightsBackendData) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      this.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+
+  publish(): Promise<void> {
+    let requestUrl = (
+      '/createhandler/status/' + this.explorationDataService.explorationId);
+
+    return this.httpClient.put(requestUrl, {
+      make_public: true
+    }).toPromise().then((response: ExplorationRightsBackendData) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      this.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    });
+  }
+
+  saveModeratorChangeToBackendAsync(emailBody: string): Promise<void> {
+    let explorationModeratorRightsUrl = (
+      '/createhandler/moderatorrights/' +
+      this.explorationDataService.explorationId);
+
+    return this.httpClient.put(explorationModeratorRightsUrl, {
+      email_body: emailBody,
+      version: this.explorationDataService.data.version
+    }).toPromise().then((response: ExplorationRightsBackendData) => {
+      let data = response;
+      this.alertsService.clearWarnings();
+      this.init(
+        data.rights.owner_names, data.rights.editor_names,
+        data.rights.voice_artist_names, data.rights.viewer_names,
+        data.rights.status, data.rights.cloned_from,
+        data.rights.community_owned, data.rights.viewable_if_private);
+    }).catch(() => {
+      this.init(
+        undefined, undefined, undefined, undefined,
+        undefined, undefined, undefined, undefined);
```

Comment:
Try to use `null` instead of `undefined`.

Reply:
Done